CARDS-2785: Form pagination should hide non-satisfied conditional sections

acrowthe opened the pull request:
- When a fetch occurs, update the pagination to hide non-satisfied conditionals
- Update the progress bar to have no "buffer" marking the current page

This is easiest to test with the Prems Outpatient Clinic form (not included in this branch) due to it's short length and conditional sections. However, this can be tested with any paginated form that contains pages which can be hidden or visible based on conditional states

acrowthe committed:
CARDS-2785: Form pagination should hide non-satisfied conditional sections
- When a fetch occurs, update the pagination to hide non-satisfied conditionals

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/FormPagination.jsx:
```diff
@@ -77,13 +77,33 @@ function FormPagination (props) {
+      if (page.canBeVisible) {
+        newNumberOfSteps++;
+      }
+    })
```
Can we make it less verbose?

```suggestion
    pages.forEach(page => page.canBeVisible && newNumberOfSteps++)
```

marta- replied:
Or even
```suggestion
    setNumberOfSteps(pages.filter(page => page.canBeVisible).length);
```

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/FormPagination.jsx:
```diff
@@ -77,13 +77,33 @@ function FormPagination (props) {
+      }
+    }
+    setNumberOfSteps(newNumberOfSteps);
+    setActiveStep(newActiveStep);
```
Can we make it less verbose?

```suggestion
    setNumberOfSteps(pages.filter(page => page.canBeVisible).length);
    setActiveStep(pages.slice(0, activePage).filter(page => page.canBeVisible).length);
```

NOTE: The suggested code has NOT been tested.

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/FormPagination.jsx:
```diff
@@ -265,23 +293,15 @@ function FormPagination (props) {
             progress: {
-              classes: {
-                bar2Buffer: classes.formStepperBufferBar,
-                dashed: classes.formStepperBackgroundBar,
```
Also remove the styling from [QuestionnaireStyle.jsx](https://github.com/data-team-uhn/cards/blob/1e1ffe3028146b99a08a943344c236c2864aad40/modules/data-entry/src/main/frontend/src/questionnaire/QuestionnaireStyle.jsx#L499-L508)

acrowthe replied:
Handled in #2128

acrowthe committed:
CARDS-2813: Improve progress bar progression
- Remove buffer for current page
- Add a half page pagging to the start and end to show some progress on the first page and when saving

acrowthe committed:
CARDS-2813: Improve progress bar progression
- Remove unused styles
- Adjust how stub size is defined

acrowthe committed:
CARDS-2785: Form pagination should hide non-satisfied conditional sections
- Clean up current and max step calculation

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/FormPagination.jsx:
```diff
@@ -213,15 +220,15 @@ function FormPagination (props) {
 
   useEffect(() => {
-    if (activePage != null && pages != null) {
+    if (activeStep != null && numberOfSteps != null) {
```
```suggestion
    if (activeStep != null && numberOfSteps > 0) {
```
^ Still ensures numberOfSteps is defined and also avoids division by 0

acrowthe committed:
CARDS-2813: Improve progress bar progression
Fix divide by 0 error leading ot progress bar initializing at 100%

acrowthe committed:
CARDS-2813: Improve progress bar progression
Rename variable
